dev/connect-gen-protos.sh

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env bash
+
+#
+# Copyright (2024) The Delta Lake Project Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+set -ex
+
+if [[ $# -gt 1 ]]; then
+  echo "Illegal number of parameters."
+  echo "Usage: $0 [path]"
+  exit -1
+fi
+
+DELTA_HOME="$(cd "`dirname $0`"/..; pwd)"
+cd "$DELTA_HOME"
+echo $DELTA_HOME
+
+OUTPUT_PATH=${DELTA_HOME}/python/delta/connect/proto/
+if [[ $# -eq 1 ]]; then
+  rm -Rf $1
+  mkdir -p $1
+  OUTPUT_PATH=$1
+fi
+
+pushd ${DELTA_HOME}/spark-connect/common/src/main
+
+LICENSE=$(cat <<'EOF'
+#
+# Copyright (2024) The Delta Lake Project Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+EOF)
+echo "$LICENSE" > /tmp/tmp_licence
+
+
+# Delete the old generated protobuf files.
+rm -Rf gen
+
+# Now, regenerate the new files
+buf generate --debug -vvv
+
+# We need to edit the generate python files to account for the actual package location and not
+# the one generated by proto.
+for f in `find gen/proto/python/delta/connect -name "*.py*"`; do
+  # First fix the imports.
+  if [[ $f == *_pb2.py || $f == *_pb2_grpc.py ]]; then
```

Comment:
This is the only part we modify the generated code. How do we know the replacements are correct? Perhaps add that to "How was this patch tested section"?

Reply:
I added a Python script to check if the generated codes are out of sync with current proto message.

In the future, when we set up the Delta Connect server and client, we will have tests for that, and if the generated codes are wrong then it will break those Delta Connect tests